### DISCUSSION TITLE
Support for DVC import dependencies

### DISCUSSION THREAD

**Discussion by sisp** [aquasecurity/trivy#4232] [Ideas]
I'd like to be able to generate SBOM for local projects which have *data dependencies* managed via [DVC](https://dvc.org/). DVC is a popular project in the machine learning (ML) community which, among other things, manages data via Git (similar to Git LFS) and supports importing data from other Git repos (similar to Git submodules) in which case a [`.dvc` file](https://dvc.org/doc/user-guide/project-structure/dvc-files#dvc-files) with a [dependency entry](https://dvc.org/doc/user-guide/project-structure/dvc-files#dependency-entries) is created. The main use case for data imports is the decoupling of data (typically ML datasets for training and testing) and ML projects (ML algorithms trained and tested on datasets) by managing data in [data registries](https://dvc.org/doc/use-cases/data-registry) and importing them in ML projects. The data in these data registries should have a license under whose terms it may be used and, thus, it would be beneficial to include this information in the SBOM.

Would you be open to receiving a contribution to Trivy (and the `go-dep-parser` library) that adds support for `.dvc` files with dependency entries?

**Comment by nejch** [comment]
This is somewhat related to https://github.com/aquasecurity/trivy/issues/3067 and all the related draft PRs, feel free to chime in there @sisp :)

**Comment by sisp** [comment]
Thanks, @nejch! :+1:
